### PR TITLE
HISTO-31 - Unified scenes for a checkpoint. Minor refactor of UnityActivity call methods.

### PR DESCRIPTION
**Pull request by fedemoglia**
Changes on **BackToGuides.cs** are irrelevant because is not more used now.

### CODE REVIEW

File: Assets/Vuforia/Scripts/Histora/SceneSwitcher.cs [hunk 1]
```diff
@@ -8,13 +8,21 @@ public class SceneSwitcher : MonoBehaviour {
...
-        ProjectVars.Instance.targetSceneName = currentScene["currentCheckpointId"].Value;
-        Debug.Log("Loading " + "guide-" + currentScene);
-        string guidesScene = "guide-" + currentScene["currentCheckpointId"].Value;
-        SceneManager.LoadScene(guidesScene);
+        var targetSceneName = currentScene["currentCheckpointId"].Value;
+        Debug.Log("Loading " + "guide-" + targetSceneName);
+        string guidesScene = "guide-" + targetSceneName;
+        /*Load the scene if exists, if not returns to AndroidApp*/
+        Debug.Log("Scene:" + SceneManager.GetSceneByName(guidesScene).ToString());
+        /*if (SceneManager.GetSceneByName(guidesScene).IsValid())
```
Comment (santiagons) [inline Assets/Vuforia/Scripts/Histora/SceneSwitcher.cs]: Would this validation be unnecessary now?

File: Assets/Vuforia/Scripts/Histora/ProjectVars.cs [hunk 2]
```diff
@@ -1,10 +1,9 @@
 ﻿public class ProjectVars : Singleton<ProjectVars> {
 
 	//Variables para instanciar
```
Comment (santiagons) [inline Assets/Vuforia/Scripts/Histora/ProjectVars.cs]: Did you mean "variables to instance"? :laughing:

File: Assets/Vuforia/Scripts/Histora/ProjectVars.cs [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
-	public string targetSceneName;
+	public string targetSceneName = "c-unlam-1";
```
Comment (santiagons) [inline Assets/Vuforia/Scripts/Histora/ProjectVars.cs]: Is there any chance of this string not being overwritten, thus having "c-unlam-1" as the target scene name in an unrelated campaign?

File: Assets/Vuforia/Scripts/Histora/SceneSwitcher.cs (hunk 1 above)
Comment (fedemoglia) [inline Assets/Vuforia/Scripts/Histora/SceneSwitcher.cs]: @santiagons It was added in this last commit. The idea was to quit Unity app if try to enter to a non existent scene and avoid the app stays freeze. But the "if statement" returns always false.

File: Assets/Vuforia/Scripts/Histora/ProjectVars.cs (hunk 2 above)
Comment (fedemoglia) [inline Assets/Vuforia/Scripts/Histora/ProjectVars.cs]: Hahaha, sorry! This class is not used now.

File: Assets/Vuforia/Scripts/Histora/ProjectVars.cs (hunk 3 above)
Comment (fedemoglia) [inline Assets/Vuforia/Scripts/Histora/ProjectVars.cs]: No, never!

**Comment by fedemoglia** [comment]
Generated a new pull request with minor changes added.

**Pull request closed without merging**